src/main/java/net/patchworkmc/patcher/util/ResourceDownloader.java

```diff
@@ -113,6 +117,54 @@ public IMappingProvider setupAndLoadMappings(Path voldemapBridged) throws IOExce
 		return bridged;
 	}
 
+	/**
+	 * Purges SRG mappings for classes that don't actually exist in the merged Minecraft jar
+	 *
+	 * <p>Newer versions of MCPConfig include mappings for classes (currently, "afg" and "dew" in 1.16.4) that do not
+	 * exist in either the Minecraft client or server jars for 1.16.4. It's unclear why exactly these mappings are
+	 * present, but interestingly enough, they are also present in the 1.16.4 Mojang mappings (client.txt / server.txt).
+	 * It seems likely that whatever automated tools the MCPConfig/Forge team use to update their mappings get confused
+	 * by this and retain the orphaned mappings.</p>
+	 *
+	 * <p>Unfortunately, some of our code doesn't particularly like these missing mappings. The code that combines the
+	 * SRG mappings (official->srg) and intermediary mappings (official->intermediary) into bridged mappings
+	 * (srg->intermediary) in particular has issues when it tries to locate the orphaned mappings within intermediary,
+	 * since intermediary doesn't contain the orphaned mappings. Therefore, we remove the orphaned mappings by checking
+	 * whether each given class actually exists in the Minecraft jar, so that they don't cause any problems.</p>
+	 */
+	private void purgeNonexistentClassMappings(List<TsrgClass<RawMapping>> classes, Path mergedMinecraftJar) throws URISyntaxException, IOException {
+		// I'm using an iterator here since it allows us to remove entries easily as we iterate over the List
+		Iterator<TsrgClass<RawMapping>> classIterator = classes.iterator();
+
+		URI jarUri = new URI("jar:" + mergedMinecraftJar.toUri());
+
+		boolean needsClarificationMessage = false;
+
+		try (FileSystem fs = FileSystems.newFileSystem(jarUri, Collections.emptyMap())) {
```

Comment:
Isn't it possible to pass `mergedMinecraftJar` in place of the URI option?

Reply:
Yep! Done.